### PR TITLE
Detect timestamp jumps

### PR DESCRIPTION
**Pull request by murillo128**
This PR detects jumps in the clock reset and timestamp jumps on the FrameDelayCalculator and avoid increasing the delay in that case.

### CODE REVIEW

**Comment by murillo128** [comment]
@harryz2000 @bcostdolby  please review it even if it is merged

File: src/FrameDelayCalculator.cpp [hunk 1]
```diff
@@ -48,23 +50,46 @@ std::chrono::milliseconds FrameDelayCalculator::OnFrame(uint64_t streamIdentifie
...
+	{
+		//Reset calculator, we keep the refTimestamp so it is not set to the stream that has the jump
+		Warning("-FrameDelayCalculator::OnFrame() | Timestamp clock jump detected [diff:%llu]\n",timestampDiff);
+		allEarlyStartTimeMs.reset();
+		frameArrivalInfo.erase(streamIdentifier);
+		refTime = 0ms;
+		return 0ms;
+	}
+	//If we were reseted because of a ts jump
+	else if (!refTime.count())
```
Comment (bcostdolby) [inline src/FrameDelayCalculator.cpp:80]: Not sure why this is an "else if "and not just an "if" I assume it has something to do with the refTimestamp only being reset

File: src/FrameDelayCalculator.cpp [hunk 2]
```diff
@@ -48,23 +50,46 @@ std::chrono::milliseconds FrameDelayCalculator::OnFrame(uint64_t streamIdentifie
...
 	bool allEarly = false;
-	if (lateMs > updateRefsPacketLateThresholdMs)  // Packet late
+	//If we detect a clock difference between the timestamps
+	if (refTimestamp && timestampDiff > MaxClockDesync)
+	{
+		//Reset calculator, we keep the refTimestamp so it is not set to the stream that has the jump
+		Warning("-FrameDelayCalculator::OnFrame() | Timestamp clock jump detected [diff:%llu]\n",timestampDiff);
+		allEarlyStartTimeMs.reset();
+		frameArrivalInfo.erase(streamIdentifier);
+		refTime = 0ms;
```
Comment (bcostdolby) [inline src/FrameDelayCalculator.cpp:76]: I didnt fully understand why refTimestamp is not reset inside this if statement along with refTime 

If we have multiple streams and then 1 of them jumps back (or forward), we are resetting the state but should we reset it for all of them. I.e. The expectation is that they all are roughly in sync as that is what it expects. 

Maybe more comments on why? The comment makes it clear there is a reason I dont know

File: src/FrameDelayCalculator.cpp (hunk 1 above)
Comment (murillo128) [reply src/FrameDelayCalculator.cpp:80]: as the `if` is doing a `return 0`  the `else if` could be an  `if``, there should be no difference though

File: src/FrameDelayCalculator.cpp (hunk 2 above)
Comment (murillo128) [reply src/FrameDelayCalculator.cpp:76]: we don't want to reset the `refTimestamp` as if not the next frame will set which clock is valid. Keeping `refTimestamp` value means that we only set the diff to 0 for the stream that has the clock jump

**Pull request merged**